pico_project.py

```diff
@@ -1128,127 +1128,134 @@ def generateProjectFiles(projectPath, projectName, sdkPath, projects, debugger):
+            "postRestartCommands": [
+                "break main",
+                "continue"
+            ],
+            "openOCDLaunchCommands": [
+                "adapter speed 1000"
+            ]
+        }},
+        {{
+            "name": "Pico Debug (Cortex-Debug with external OpenOCD)",
+            "cwd": "${{workspaceRoot}}",
+            "executable": "${{command:cmake.launchTargetPath}}",
+            "request": "launch",
+            "type": "cortex-debug",
+            "servertype": "external",
+            "gdbTarget": "localhost:3333",
+            "gdbPath": "{gdbPath}",
+            "device": "RP2040",
+            "svdFile": "${{env:PICO_SDK_PATH}}/src/rp2040/hardware_regs/rp2040.svd",
+            "runToEntryPoint": "main",
+            // Give restart the same functionality as runToEntryPoint - main
+            "postRestartCommands": [
+                "break main",
+                "continue"
+            ]
+        }},
+        {{
+            "name": "Pico Debug (C++ Debugger)",
+            "type": "cppdbg",
+            "request": "launch",
+            "cwd": "${{workspaceRoot}}",
+            "program": "${{command:cmake.launchTargetPath}}",
+            "MIMode": "gdb",
+            "miDebuggerPath": "{gdbPath}",
+            "miDebuggerServerAddress": "localhost:3333",
+            "debugServerPath": "openocd",
+            "debugServerArgs": "-f interface/cmsis-dap.cfg -f target/rp2040.cfg -c \\"adapter speed 1000\\"",
+            "serverStarted": "Listening on port .* for gdb connections",
+            "filterStderr": true,
+            "stopAtEntry": true,
+            "hardwareBreakpoints": {{
+                "require": true,
+                "limit": 4
+            }},
+            "preLaunchTask": "Flash",
+            "svdPath": "${{env:PICO_SDK_PATH}}/src/rp2040/hardware_regs/rp2040.svd"
+        }},
+    ]
+}}
+'''
+
+            properties = f'''{{
+    "configurations": [
+        {{
+            "name": "Pico",
+            "includePath": [
+                "${{workspaceFolder}}/**",
+                "${{env:PICO_SDK_PATH}}/**"
+            ],
+            "defines": [],
+            "compilerPath": "{cPath}",
+            "cStandard": "c17",
+            "cppStandard": "c++14",
+            "intelliSenseMode": "linux-gcc-arm",
+            "configurationProvider": "ms-vscode.cmake-tools"
+        }}
+    ],
+    "version": 4
+}}
+'''
+
+
+            # settings
+            settings = f'''{{
+    "cmake.statusbar.advanced": {{
+        "debug": {{
+            "visibility": "hidden"
+        }},
+        "launch": {{
+            "visibility": "hidden"
+        }},
+        "build": {{
+            "visibility": "hidden"
+        }},
+        "buildTarget": {{
+            "visibility": "hidden"
+        }}
+    }},
+    "cmake.buildBeforeRun": true,
+    "cmake.configureOnOpen": true,
+    "cmake.configureSettings": {{
+        "CMAKE_MODULE_PATH": "${{env:PICO_INSTALL_PATH}}/pico-sdk-tools"
```

Comment:
Just a note, PICO_INSTALL_PATH only exists on Windows when the installer is used. Should be harmless otherwise but should be tested I think.